sql/stripe/funnel/view.sql

```diff
@@ -0,0 +1,65 @@
+CREATE OR REPLACE VIEW
+  funnel
```

Comment:
r? @felixlawrence on this view

My goal here is to give a useful representation of the `subscribed` stage of the funnel, with room to add other stages later.

For context, the rest of the queries here (except the revenue view) are just to extract the most up to date values we have for each resource type, and hide any fields that shouldn't be accessible to everyone in re:dash.

Reply:
I think I don't understand what funnel we're trying to represent here.

The usual way I think of a funnel is as a series of ~tubes~ steps. Each of the steps has a starting event, and a range of possible outcomes. For each outcome, we can calculate the conversion rate towards that outcome.

When writing queries to analyse a funnel, I'd typically write a query for each step: start with a list of the starting events in a given time window, typically `LEFT JOIN` to the corresponding outcome event to see what the outcome was (limiting the conversion window to some fixed period of time, to avoid lagging metrics where history constantly gets rewritten), and optionally apply a `DISTINCT user_id` to the count of starting events (denominator) and the count of each outcome (numerator).

I can see how the `subscriptions_v1` table has already done some of the `LEFT JOIN`ing for us...

I can see how "start entering payment details" -> "successfully set up payment" is a one-step funnel, and "start entering payment details" -> "successfully set up payment" -> "renew for your second month" is a two-step funnel.

I can see a few funnels being relevant to the VPN product:
* Arrives on landing page -> clicks CTA on landing page -> signs up/in to a FxA -> ... -> Adds payment method -> first payment (or CC auth?) successful -> renews for 2nd month -> renews for 3rd month
* Renews for nth month -> renews for n+1th month
* Subscribes -> runs client app -> runs client app on 2nd device -> runs client app on 3rd device
* Has payment failure -> resolves payment failure

This view is, for each date, sorting subscriptions by their status and counting them, which doesn't match my understanding of funnel analyses - so it would be good to sync on Monday about what the underlying objectives here are, so I'd be better placed to review whether it's meeting them!